src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -4682,7 +4682,15 @@ private BoundExpression MakeBadExpressionForObjectCreation(SyntaxNode node, Type
                     break;
 
                 case BoundKind.PropertyAccess:
-                    hasErrors |= isRhsNestedInitializer && !CheckNestedObjectInitializerPropertySymbol(((BoundPropertyAccess)boundMember).PropertySymbol, namedAssignment.Left, diagnostics, hasErrors, ref resultKind);
+                    var propertySymbol = ((BoundPropertyAccess)boundMember).PropertySymbol;
+                    var memberNameSyntax = namedAssignment.Left;
+                    hasErrors |= isRhsNestedInitializer && !CheckNestedObjectInitializerPropertySymbol(propertySymbol, memberNameSyntax, diagnostics, hasErrors, ref resultKind);
+
+                    if (propertySymbol.SetMethod?.IsInitOnly == true)
+                    {
+                        CheckFeatureAvailability(memberNameSyntax, MessageID.IDS_FeatureInitOnlySetters, diagnostics);
```

Comment:
>CheckFeatureAvailability(memberNameSyntax, MessageID.IDS_FeatureInitOnlySetters, diagnostics); [](start = 24, length = 94)

I would expect this check to be performed next to where we report or don't report ERR_AssignmentInitOnly. Otherwise, we are likely to miss other scenarios where usage of init-only setters in allowed, in constructors, for example.

Reply:
Thanks. That makes sense

---
In reply to: [552032297](https://github.com/dotnet/roslyn/pull/50251#discussion_r552032297) [](ancestors = 552032297)